libsel4vm/src/arch/arm/vgic/vgic.c

```diff
@@ -467,56 +469,42 @@ static inline int vgic_add_overflow_cpu(struct lr_of *lr_overflow, struct virq_h
     return 0;
 }
 
-static inline int vgic_add_overflow(vgic_t *vgic, struct virq_handle *irq, vm_vcpu_t *vcpu)
+static inline void vgic_handle_overflow(vgic_t *vgic, vm_vcpu_t *vcpu)
```

Comment:
Could call this `vgic_process_lr_queue` now, because this is what is is doing effectively

Reply:
Actually, why not fully inline this completely, the maintenance routine is the only place that calls this

Reply:
Please check my new approach; personally I don't like inlining, I want to hide the implementation details especially in those are completely orthogonal to the workings of the caller.